### DISCUSSION TITLE
Allow dot (. or ./) as basePath

### DISCUSSION THREAD

**Discussion by kirill-konshin** [vercel/next.js#17544] [Ideas]
# Bug report

## Describe the bug

Next.js does not allow anything besides paths starting with slash as `basePath`.

## To Reproduce

When `next.config.js` is as follows:

```js
module.exports = {
  basePath: './',
};
```

then on build you get an error:

```
> Build error occurred
Error: Specified basePath has to start with a /, found "./"
```

## Expected behavior

Dot as `basePath` is perfectly legit since files can be taken from where `index.html` was loaded in cases when no navigation is used.

I am using Next.js to build an app for Electron, and there the app would be served with `file://` protocol. And the problem is that final location of app is not quite determined...

I can get around this by string replacing the path directly in the `out` directory inside the electron app, but that's nasty...

## System information

- OS: macOS
- Version of Next.js: 9.5.3
- Version of Node.js: 14.4.0

**Comment by lfades** [comment]
@kirill-konshin From the [Base Path docs](https://nextjs.org/docs/api-reference/next.config.js/basepath):

> To deploy a Next.js application under a sub-path of a domain you can use the `basePath` config option.
>
> `basePath` allows you to set a path prefix for the application.

`basePath` is nothing more than a prefix, it wasn't never intended for it to do any kind of file resolution.

**Comment by kirill-konshin** [comment]
@lfades I know, I've read the docs. And the point is that I want `href="_next/whatever"` or `href="./_next/whatever"` in my `out/index.html`. And Next.js only allows to set something with starting slash: `href="/_next/whatever"`. Create react app allows to set `PUBLIC_URL=./`, for example...

The only problem is that error message `Specified basePath has to start with a /, found "./"` which does not allow legit path prefix...

**Comment by timneutkens** [comment]
You're referring to `assetPrefix` which is there to configure the prefixed path for files.

**Reply by kirill-konshin** [reply]
You're right, it worked. A bit misleading, but it's OK! Thank you for your time.

**Comment by ganeshsacharya** [comment]
For all who reached here, i had the same situation and after reading the above comments changed my next.config.js to below.
```javascript
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  assetPrefix: process.env.BASE_PATH
}
```
My project is a statically exported NextJS website for which i use the below build commands.
```
"build": "BASE_PATH=./ npm run build-2"
"build-2": "next build && next export"
```

Hope you find this useful. Thanks for the discussions.
